2017/3/day3_2.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"fmt"
+)
+
+type point struct {
+	X int
+	Y int
+}
+
+func (p point) String() string {
+	return fmt.Sprintf("(%v, %v)", p.Y, p.X)
+}
+
+func nextPoint(subtract bool, coord string, p point) point {
+	if coord == "x" {
+		if subtract {
+			return point{p.X - 1, p.Y}
+		}
+		return point{p.X + 1, p.Y}
+	}
+
+	// y case
+	if subtract {
+		return point{p.X, p.Y - 1}
+	}
+	return point{p.X, p.Y + 1}
+}
+
+func nextVal(m map[point]int, loc point) int {
+	var neighbours = []point{
+		point{
+			Y: -1,
+			X: -1,
+		},
+		point{
+			Y: -1,
+			X: 0,
+		},
+		point{
+			Y: -1,
+			X: 1,
+		},
+		point{
+			Y: 0,
+			X: -1,
+		},
+		point{
+			Y: 0,
+			X: 1,
+		},
+		point{
+			Y: 1,
+			X: -1,
+		},
+		point{
+			Y: 1,
+			X: 0,
+		},
+		point{
+			Y: 1,
+			X: 1,
+		},
```

Comment:
@BlueMonday do you know of a better way to declare these? In python you can actually do something crazy like:

```
list(itertools.product([-1, 0, 1], [-1, 0, 1]))
```

Reply:
I don't believe there is anything like `itertools.product`. If you want it to be more concise, I think you could do it with two for loops?

Reply:
ah kk, i feel like 2 for loops would be even harder to read